canopen/sdo/client.py

```diff
@@ -487,14 +487,14 @@ def __init__(self, sdo_client, index, subindex=0, request_crc_support=True):
         if res_command & 0xE0 != RESPONSE_BLOCK_UPLOAD:
             self._error = True
             self.sdo_client.abort(0x05040001)
-            raise SdoCommunicationError("Unexpected response 0x%02X" % res_command)
+            raise SdoCommunicationError(f"Unexpected response 0x{res_command:02X}")
         # Check that the message is for us
         if res_index != index or res_subindex != subindex:
             self._error = True
-            raise SdoCommunicationError((
-                "Node returned a value for 0x{:X}:{:d} instead, "
+            raise SdoCommunicationError(
+                f"Node returned a value for 0x{res_index:X}:{res_subindex:d} instead, "
```

Comment:
Should use hex here as well for the subindex. As previously discussed.

Reply:
Agreed. Will be updated.